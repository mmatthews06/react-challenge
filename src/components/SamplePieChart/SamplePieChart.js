import React from 'react';
import PropTypes from 'prop-types';

import {
  Cell,
  Legend,
  ResponsiveContainer,
  Pie,
  PieChart,
} from 'recharts';

const COLORS = ['#0088FE', '#00C49F', '#FFBB28', '#FF8042'];

export default function SamplePieChart({ active, queued, onHold }) {
  const total = active + queued + onHold;
  const data = [
    { label: 'Active', value: active },
    { label: 'Queued', value: queued },
    { label: 'Active', value: onHold },
  ];

  if (total === 0) {
    return <span>No data to show!</span>;
  }

  return (
    <ResponsiveContainer>
      <PieChart>
        <Pie
          data={data}
          dataKey="value"
          nameKey="label"
          innerRadius="50%"
          fill="#8884d8"
          label
        >
          {
            data.map((entry, index) => <Cell key={`cell-${entry.label}`} fill={COLORS[index % COLORS.length]} />)
          }
        </Pie>
        <Legend verticalAlign="top" />
      </PieChart>
    </ResponsiveContainer>
  );
}

SamplePieChart.propTypes = {
  active: PropTypes.number.isRequired,
  queued: PropTypes.number.isRequired,
  onHold: PropTypes.number.isRequired,
};
