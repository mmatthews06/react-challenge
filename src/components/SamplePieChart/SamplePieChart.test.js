/* global describe, expect, it */
import React from 'react';
import { shallow } from 'enzyme';

import SamplePieChart from './SamplePieChart';

describe('SamplePieChart', () => {
  it('should render an empty message', () => {
    expect(shallow(<SamplePieChart active={0} queued={0} onHold={0} />)).toMatchSnapshot();
  });

  it('should match a snapshot', () => {
    expect(shallow(<SamplePieChart active={20} queued={10} onHold={3} />)).toMatchSnapshot();
  });
});
