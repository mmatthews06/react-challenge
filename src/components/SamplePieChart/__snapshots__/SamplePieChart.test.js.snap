// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SamplePieChart should match a snapshot 1`] = `
<ResponsiveContainer
  debounce={0}
  height="100%"
  width="100%"
>
  <PieChart
    barCategoryGap="10%"
    barGap={4}
    cx="50%"
    cy="50%"
    endAngle={360}
    innerRadius={0}
    layout="centric"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    outerRadius="80%"
    reverseStackOrder={false}
    stackOffset="none"
    startAngle={0}
  >
    <Pie
      animationBegin={400}
      animationDuration={1500}
      animationEasing="ease"
      blendStroke={false}
      cx="50%"
      cy="50%"
      data={
        Array [
          Object {
            "label": "Active",
            "value": 20,
          },
          Object {
            "label": "Queued",
            "value": 10,
          },
          Object {
            "label": "Active",
            "value": 3,
          },
        ]
      }
      dataKey="value"
      endAngle={360}
      fill="#8884d8"
      hide={false}
      innerRadius="50%"
      isAnimationActive={true}
      label={true}
      labelLine={true}
      legendType="rect"
      minAngle={0}
      nameKey="label"
      outerRadius="80%"
      paddingAngle={0}
      startAngle={0}
      stroke="#fff"
    />
  </PieChart>
</ResponsiveContainer>
`;

exports[`SamplePieChart should render an empty message 1`] = `
<span>
  No data to show!
</span>
`;
